listener.go

```diff
@@ -33,11 +34,22 @@ type listener struct {
 // serverPrivateKey - if provided, this listener will expect connections to use
 //                    encryption
 func WrapListener(wrapped net.Listener, pool BufferPool, serverPrivateKey *rsa.PrivateKey) net.Listener {
+	return WrapListenerWithErrorHandler(wrapped, pool, serverPrivateKey, nil)
+}
+
+// WrapListenerWithErrorHandler is like WrapListener and also supports a
+// callback for errors on acceptnig new connections.
```

Comment:
acceptnig

Reply:
Implemetned :)

I also renamed this function to `WrapListenerIncludingErrorHandler` since `WrapListenerWithErrorHandler` sounds like we're wrapping a listener with an error handler, which isn't really what we're doing here.